src/app/component/projects/pages/harvest/harvest.component.ts

```diff
@@ -0,0 +1,177 @@
+import { WidgetMenuItem } from "src/app/component/shared/widget/widgetItem";
+import { Page } from "src/app/helpers/page/pageDecorator";
+import { Project } from "src/app/models/Project";
+import { projectResolvers } from "src/app/services/baw-api/projects.service";
+import { ResolvedModel } from "src/app/services/baw-api/resolver-common";
+import {
+  harvestProjectMenuItem,
+  projectCategory,
+  projectMenuItem,
+} from "../../projects.menus";
+import { projectMenuItemActions } from "../details/details.component";
+
+const projectKey = "project";
+
+@Page({
+  category: projectCategory,
+  menus: {
+    actions: List([projectMenuItem, ...projectMenuItemActions]),
+    actionsWidget: new WidgetMenuItem(PermissionsShieldComponent, {}),
+    links: List(),
+  },
+  resolvers: {
+    [projectKey]: projectResolvers.show,
+  },
+  self: harvestProjectMenuItem,
+})
+@Component({
+  selector: "app-harvest",
+  templateUrl: "./harvest.component.html",
+  styleUrls: ["./harvest.component.scss"],
+  // tslint:disable-next-line: use-component-view-encapsulation
+  encapsulation: ViewEncapsulation.None,
+})
+export class HarvestComponent implements OnInit {
+  public buttons: {
+    previous: { disabled?: boolean; text?: string };
+    next: { disabled?: boolean; text?: string };
+  };
+  public failure: boolean;
+  public filesize = filesize;
+  public harvest = Harvest;
+  public progress: number;
+  public project: Project;
+  public stage: Harvest;
+  public stages: MenuItem[] = [
+    { label: "Start" },
+    { label: "Credentials" },
+    { label: "Check" },
+    { label: "Review" },
+    { label: "Harvest" },
+    { label: "Summary" },
+  ];
+  private interval: number;
+  private intervalSpeed = 300;
+
+  constructor(private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    const resolvedProject: ResolvedModel<Project> = this.route.snapshot.data[
+      projectKey
+    ];
+
+    if (resolvedProject.error) {
+      this.failure = true;
+      return;
+    }
+
+    this.project = resolvedProject.model;
+    this.stage = Harvest.Start;
+    this.updateNavigation();
+  }
+
+  public nextStage() {
+    this.stage++;
+    clearInterval(this.interval);
+    this.updateNavigation();
+  }
+
+  public previousStage() {
+    // Review page should go to Credentials
+    if (this.stage === Harvest.Review) {
+      this.stage = Harvest.Credentials;
+    } else {
+      this.stage--;
+    }
+
+    clearInterval(this.interval);
+    this.updateNavigation();
+  }
+
+  private mockTimer(callback?: () => void) {
+    this.progress = 0;
+
+    // https://github.com/TypeStrong/atom-typescript/issues/1053
+    this.interval = (setInterval as any)(() => {
+      this.progress++;
+
+      if (this.progress >= 100) {
+        this.progress = 100;
+        clearInterval(this.interval);
+
+        if (callback) {
+          callback();
+        }
+      }
+    }, this.intervalSpeed);
+  }
+
```

Comment:
Could probably use rxjs http://reactivex.io/documentation/operators/interval.html to emit event periodically.

While mocking, we'd generate fake progress.
When stubbing to a real service we'd then do an api call, but the same observable mechanism can be used

Reply:
I have changed the system to utilize an observable now, however i have not extracted it to a harvest service for fear of a merge conflict with the changes we have made to the api services.